core/common.go

```diff
@@ -240,25 +240,22 @@ func buildFindLocalImageOptions(image string) docker.ListImagesOptions {
 }
 
 func buildPullOptions(image string) (docker.PullImageOptions, docker.AuthConfiguration) {
-	tag := "latest"
-	registry := ""
-
-	parts := strings.Split(image, ":")
-	if len(parts) == 2 {
-		tag = parts[1]
-	}
-
-	name := parts[0]
-	parts = strings.Split(name, "/")
-	if len(parts) > 2 {
-		registry = parts[0]
-	}
-
-	return docker.PullImageOptions{
-		Repository: name,
-		Registry:   registry,
-		Tag:        tag,
-	}, buildAuthConfiguration(registry)
+        re := regexp.MustCompile("((?:([^/]+\\.[^/]+)/)?[^:]+)(?::(.+))?")
+        matches := re.FindStringSubmatch(image)
+
+        repository := matches[1]
```

Comment:
You need to check for `len(matches)` to make sure the code will not panic if someone will provide a bad image. Like empty string, etc.

Reply:
This will not be needed if we will use the method from this comment https://github.com/mcuadros/ofelia/pull/119#pullrequestreview-504060337

Reply:
Okay that's fine.  Couldn't see any code in there to extract the registry from the repository so will still need something to pull this out.  Is a port actually a required part of the registry? e.g. localhost:5000/myimage:tag

If so then it should be fairly simple.

Reply:
This is a good question about the port for the registry. I don't know...
Probably it is needed.

Does the registry you use require auth? If so, can you check the `~/.docker/config.json` in which format your registry is stored?

Reply:
Yes it requires auth and my config.json is in auth:{ } format.

I'm working on the changes, but this is now an issue:
https://github.com/mcuadros/ofelia/blob/master/core/runjob_test.go#L94
This test here does not have a port and not sure if to change or not?

Reply:
> This test here does not have a port and not sure if to change or not?
Does that test pass? I think it is fine without a port since it is tested in the `go-dockerclient` lib.

> Yes it requires auth and my config.json is in auth:{ } format.
Are there your registry key inside `auth{}` object?

For example mine like like:
```json
  "auths" : {
    "https://index.docker.io/v1/": {
         ...
    }
  }
```

Just wondering if yours has port in there.

Reply:
But yeah, if in this example registry URL contains `/v1/` path, then I assume, that `registry` should be with the port.

Reply:
```
"auths": {
  "gitlab.xxx.xx.xx:4567": {
    ...
  }
}
```

is how mine appears

Reply:
Thanks, this is what I wanted to see.
This confirms that port is needed as part of `registry`!

Reply:
I've made the changes and separated the registry parsing into another function for easier testing.  The test I identified above is failing and not sure what the expected behaviour should be.

https://stackoverflow.com/questions/44880429/docker-push-to-private-registry-how-not-to-specify-a-port seems to suggests that you _can_ have a registry with no port.  In that case I am not sure how you work out whether `server/image` is the image `image` on the server `server` or a local image called `server/image`.  Maybe I just don't understand how the naming works...

Maybe we should just be searching for prefixes in the auths and using that should we get a prefix match.

That said, I still don't know how compatible this is with your config file and `https://index.docker.io/v1/` in the `auths`

Reply:
Thanks, I will check the failing test later

Reply:
I improved a bit `parseRegistry` function. Thanks